src/tardev-snapshotter/src/snapshotter.rs

```diff
@@ -1205,8 +1268,8 @@ impl TarDevSnapshotter {
             let to = store
                 .layer_path_to_write(&key)
                 .map_err(|e| Status::unknown(format!("failed to create layer path: {e}")))?;
-            trace!("Renaming from {:?} to {:?}", &snapshot_name, &to);
-            tokio::fs::rename(snapshot_name, to).await?;
+            trace!("Renaming from {:?} to {:?}", &erofs_snapshot_name, &to);
```

Comment:
I am now renaming erofs_snapshot_name (which is base_name + .erofs). Therefore base_name still exists at this point. Before snapshot_name was being renamed (which was base_name). Is this an issue?

Reply:
Isnt the file erofs+merkle tree?

Reply:
Separately to your comment, the original logic I think tried to limit extra files/writes. At minimum, we should delete the unused files. Better, we should not produce unnecessary copies, though not sure if that is possible here.